charts/tempo/templates/networkpolicy.yaml

```diff
@@ -0,0 +1,61 @@
+{{- if .Values.networkPolicy.enabled }}
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: {{ include "tempo.fullname" . }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    {{- include "tempo.labels" . | nindent 4 }}
+    {{- with .Values.labels }}
+    {{- toYaml . | nindent 4 }}
+    {{- end }}
+  {{- with .Values.annotations }}
+  annotations:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+spec:
+  policyTypes:
+    {{- if .Values.networkPolicy.ingress }}
+    - Ingress
+    {{- end }}
+    {{- if .Values.networkPolicy.egress.enabled }}
+    - Egress
+    {{- end }}
+  podSelector:
+    matchLabels:
+      {{- include "tempo.selectorLabels" . | nindent 6 }}
+
+  {{- if .Values.networkPolicy.egress.enabled }}
+  egress:
+    {{- if not .Values.networkPolicy.egress.blockDNSResolution }}
```

Comment:
Are we able to cover this rule with the other options?  What's missing if not?

Reply:
egress is an important part of the Network Policy resource it basically defines rules for outgoing traffic from the pods.

Reply:
No sorry, my question is more specific to the blockDnsResolution.  If the primitives allow for specifying in the user values "port 53" and "protocol UDP", then this doesn't seem needed, correct?

Reply:
Something like the following in my local values would avoid needing this added option if I understand.
```yaml
networkPolicy:
  enabled: true
  egress:
    ports: 
      - port: 53
        protocol: UDP
```

Reply:
yes that would do, but the reason behind going with the below code was to maintain uniformity as the same code in the grafana NP is working fine.

```
  {{- if .Values.networkPolicy.egress.enabled }}
  egress:
    {{- if not .Values.networkPolicy.egress.blockDNSResolution }}
 ```

Reply:
And as per my understanding we are simplifying it for the users, if they want to block DNS resolution they would simply make it true, without needing to define this:
```
networkPolicy:
  enabled: true
  egress:
    ports: 
      - port: 53
        protocol: UDP
```

and if they don't want to block DNS resolution the below part would be executed allowing traffic in Port 53:

```
  {{- if .Values.networkPolicy.egress.enabled }}
  egress:
    {{- if not .Values.networkPolicy.egress.blockDNSResolution }}
```

In this case they don't have to wright a single piece of code to block DNS resolution, it would be just true or false and the rest would follow.